src/new/wallet.js

```diff
@@ -511,6 +518,151 @@ class HathorWallet extends EventEmitter {
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   **/
+  mintTokens(tokenUid, amount, address) {
+    storage.setStore(this.store);
+    const mintAddress = address || this.getCurrentAddress();
+    const walletData = wallet.getWalletData();
+    let mintInput = null;
+    for (const tx_id in walletData.historyTransactions) {
+      const tx = walletData.historyTransactions[tx_id];
+      if (tx.is_voided) {
+        // Ignore voided transactions.
+        continue;
+      }
+
+      for (const [index, output] of tx.outputs.entries()) {
+        // This output is not mine
+        if (!wallet.isAddressMine(output.decoded.address, walletData)) {
+          continue;
+        }
+
+        // This token is not the one of this screen
+        if (output.token !== tokenUid) {
+          continue;
+        }
+
+        // If output was already used, we can't use it
+        if (output.spent_by) {
+          continue;
+        }
+
+        if (wallet.isMintOutput(output)) {
+          mintInput = {tx_id, index, address: output.decoded.address};
+          break
+        }
+      }
+    }
+
+    if (mintInput === null) {
+      return {success: false, message: 'Don\'t have mint authority output available.'}
+    }
+
+    const ret = tokens.mintTokens(mintInput, tokenUid, mintAddress, amount, null, this.pinCode);
+
+    if (ret.success) {
+      const sendTransaction = ret.sendTransaction;
+      sendTransaction.start();
+      return {success: true, promise: sendTransaction.promise, sendTransaction};
+    } else {
+      return ret;
+    }
+  }
+
+  /**
+   * Melt tokens
+   *
+   * @param {String} tokenUid UID of the token to melt
+   * @param {number} amount Quantity to melt
+   *
+   * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   **/
+  meltTokens(tokenUid, amount) {
+    storage.setStore(this.store);
+    let meltInput = null;
+    const walletData = wallet.getWalletData();
+    for (const tx_id in walletData.historyTransactions) {
+      const tx = walletData.historyTransactions[tx_id];
+      if (tx.is_voided) {
+        // Ignore voided transactions.
+        continue;
+      }
+
+      for (const [index, output] of tx.outputs.entries()) {
+        // This output is not mine
+        if (!wallet.isAddressMine(output.decoded.address, walletData)) {
+          continue;
+        }
+
+        // This token is not the one of this screen
+        if (output.token !== tokenUid) {
+          continue;
+        }
+
+        // If output was already used, we can't use it
+        if (output.spent_by) {
+          continue;
+        }
+
+        if (wallet.isMeltOutput(output)) {
+          meltInput = {tx_id, index, address: output.decoded.address};
+          break
+        }
+      }
+    }
```

Comment:
It might be better to refactor this part as it shares a lot of code with the mint code. So we could just pass the functions `wallet.isMeltOutput` and `wallet. isMintOutput` as params. For instance:
```js
const meltOutput = selectAuthorityUtxo(wallet.isMeltOutput, walletData);
```

Reply:
Done!